impl/src/main/java/org/jboss/weld/util/reflection/Reflections.java

```diff
@@ -465,14 +465,20 @@ public static Exception unwrapInvocationTargetException(InvocationTargetExceptio
         }
     }
 
-    public static Set<Class<?>> getInterfaceClosure(Class<?> clazz) {
+    public static Set<Class<?>> getInterfaceClosure(Class<?> clazz, Set<Class<?>> setOfInterfaces) {
```

Comment:
We need to add a unit test to `org.jboss.weld.tests.unit.util.reflection.ReflectionsTest` to have a verification of this new behaviour.

Reply:
Something like this should work:
```java
public static Set<Class<?>> getInterfaceClosure(Class<?> clazz) {
   Set<Class<?>> result = new HashSet<>();  
   for (Class<?> classToDiscover = clazz; classToDiscover != null; classToDiscover = classToDiscover.getSuperclass()) {
     addInterfaces(classToDiscover, result);
   }
   return result;
}
```